packages/rs-dapi-client/src/executor.rs

```diff
@@ -55,12 +79,48 @@ pub struct ExecutionResponse<R> {
     pub address: Address,
 }
 
-impl<R> ExecutionResponse<R> {
+impl<R, I> IntoInner<I> for ExecutionResponse<R>
+where
+    R: Into<I>,
+{
     /// Unwrap the response
-    pub fn into_inner(self) -> R {
-        self.inner
+    fn into_inner(self) -> I {
+        self.inner.into()
+    }
+}
+
+impl<T: VersionedGrpcResponse> VersionedGrpcResponse for ExecutionResponse<T> {
+    type Error = T::Error;
+
+    fn metadata(&self) -> Result<&dapi_grpc::platform::v0::ResponseMetadata, Self::Error> {
+        self.inner.metadata()
+    }
+    fn proof(&self) -> Result<&dapi_grpc::platform::v0::Proof, Self::Error> {
+        self.inner.proof()
+    }
+    fn proof_owned(self) -> Result<dapi_grpc::platform::v0::Proof, Self::Error> {
+        self.inner.proof_owned()
+    }
+}
+
+impl<R> From<R> for ExecutionResponse<R> {
```

Comment:
This must be used only for tests, and we want to isolate it form production code.